components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/dto/SAMLSSOReqValidationResponseDTO.java

```diff
@@ -321,4 +342,71 @@ public void addAuthenticationContextClassRef(
         }
         authenticationContextClassRefList.add(authenticationContextClassRefDTO);
     }
+
+    /**
+     * Get requested attributes.
+     *
+     * @return list of requested attributes
+     */
+    public List<ClaimMapping> getRequestedAttributes() {
+
+        return requestedAttributes;
+    }
+
+    /**
+     * Set requested attributes.
+     *
+     * @param requestedAttributes list of requested attributes
+     */
+    public void setRequestedAttributes(List<ClaimMapping> requestedAttributes) {
+
+        if (this.requestedAttributes == null) {
+            this.requestedAttributes = requestedAttributes;
+        } else {
+            this.requestedAttributes.addAll(requestedAttributes);
+        }
+    }
+
+    /**
+     * Get Authentication Context Comparison.
+     *
+     * @return Authentication Context Comparison
+     */
+    public String getRequestedAuthnContextComparison() {
+
+        return requestedAuthnContextComparison;
+    }
+
+    /**
+     * Set Authentication Context Comparison.
+     *
+     * @param requestedAuthnContextComparison Authentication Context Comparison
+     */
+    public void setRequestedAuthnContextComparison(String requestedAuthnContextComparison) {
+
+        this.requestedAuthnContextComparison = requestedAuthnContextComparison;
+    }
+
+    /**
+     * Get properties.
+     *
+     * @return request properties
+     */
+    public Properties getProperties() {
+
+        if (properties == null) {
+            properties = new Properties();
+        }
+        return properties;
+    }
+
+    /**
+     * Set properties.
+     *
+     * @param properties request properties
+     */
+    public void setProperties(Properties properties) {
+
+        this.properties = properties;
+    }
```

Comment:
I think it would be better to add the capability to set a property or add a property rather than allowing set a bunch of properties.